lib/Db/MessageMapper.php

```diff
@@ -793,20 +793,59 @@ public function findByMessageId(Account $account, string $messageId): array {
 	}
 
 	/**
-	 * @param Mailbox $mailbox
-	 * @param SearchQuery $query
-	 * @param int|null $limit
 	 * @param int[]|null $uids
-	 *
 	 * @return int[]
+	 *
+	 * @throws Exception
 	 */
 	public function findIdsByQuery(Mailbox $mailbox, SearchQuery $query, string $sortOrder, ?int $limit, ?array $uids = null): array {
+		return array_map(
+			static fn (Message $message) => $message->getId(),
+			$this->findByQueryWithoutRelatedData(
+				$mailbox,
+				$query,
+				$sortOrder,
+				$limit,
+				$uids,
+			),
+		);
```

Comment:
Potential drawback: Overfetching

If only the IDs are required, the whole row is fetched without being required.

This method is currently only used in `\OCA\Mail\Service\Sync\SyncService::getDatabaseSyncChanges()`.